### ISSUE TITLE
Access token generation from refresh token fails for SF-Concur-Integration

### ISSUE THREAD

**Issue by HiranyaKavishani** [ballerina-platform/ballerina-lang#10641]
Description:
There is a  SalesForce-Concur connector endpoint.  If the access token is valid, all the requests and responses are working fine with concur endpoint.

After the access token expiration, it is not being regenerated by refresh token since it need to pass another parameter called 'scope' and ballerina auth doesn't support for that parameter.

It was worked by calling normal httpRefreshToken end point as follows.

Steps to reproduce:

string newAccessToken;

endpoint http:Client httpRefreshToken {
    url: "https://us.api.concursolutions.com"
};

endpoint http:Client httpEndpoint {
    url: "https://www.concursolutions.com/api/v3.0/common/listitems",
    auth: {
        scheme: http: OAUTH2,
        accessToken: newAccessToken,
        clientId: config:getAsString("CLIENT_ID", default = CLIENT_ID),
        clientSecret: config:getAsString("CLIENT_SECRET", default = CLIENT_SECRET),
        refreshToken: config:getAsString("REFRESH_TOKEN", default = REFRESH_TOKEN),
        refreshUrl: config:getAsString("REFRESH_URL", default = REFRESH_URL),
        grant_type: config:getAsString("GRANT_TYPE", default = CLIENT_ID),
        username: config:getAsString("USERNAME", default = USERNAME),
        password: config:getAsString("PASSWORD", default = PASSWORD)
    }
};

function refreshToken() returns error? {
    http:Request requestToken;
    requestToken.setTextPayload("client_id=" + config:getAsString("CLIENT_ID", default = CLIENT_ID) + "&client_secret=" + config:getAsString("CLIENT_SECRET", default = CLIENT_SECRET) + "&grant_type=refresh_token& refresh_token=" + config:getAsString("REFRESH_TOKEN", default = REFRESH_TOKEN) + "&scope=" + config:getAsString("SCOPE", default = SCOPE), contentType = mime:APPLICATION_FORM_URLENCODED);

    var tokenresponse = httpRefreshToken->post("/oauth2/v0/token", requestToken);
    match tokenresponse {
        http:Response resp => {
            var msg = resp.getJsonPayload();
            match msg {
                json jsonPayload => {
                    io:println(jsonPayload.access_token);
                    //string accessTokenByjson  = check <string> jsonPayload2.access_token;
                    // newAccessToken = setAccessToken(accessTokenByjson);
                }
                error err => {
                    log:printError(err.message, err = err);
                    return err;
                }
            }
        }
    }
    return();
}

Affected Versions: 0.980.0

OS, DB, other environment details and versions:

Related Issues (optional):
https://github.com/ballerina-platform/ballerina-lang/issues/10608

Suggested Labels (optional):

Suggested Assignees (optional):

**Comment by Anoukh** [comment]
@HiranyaKavishani we have identified an issue in generating the access token with a refresh token. This will be fixed in the next release.

**Comment by MadhukaHarith92** [comment]
@Anoukh I face a similar issue with wso2/gmail package (0.9.7). Will the fix mentioned by you be applied to wso2/gmail also?

**Comment by Anoukh** [comment]
Yes @MadhukaHarith92 it will be fix to all clients using OAuth2.0 for authentication
